packages/playground/windows/playground/App.cpp

```diff
@@ -0,0 +1,275 @@
+        L"       IsChecked='false'"
+        L"       Content='Break on First Line'/>"
+        L"     <CheckBox x:Name='x_UseFastRefreshCheckBox'"
+        L"       Margin='5,2'"
+        L"       VerticalAlignment='Center'"
+        L"       IsChecked='true'"
+        L"       Content='Fast Refresh'/>"
+        L"     <TextBlock"
+        L"       VerticalAlignment='Center'"
+        L"       Margin='5,0,5,3'"
+        L"       Text='Debugger Port'/>"
+        L"     <TextBox x:Name='x_DebuggerPort'"
+        L"       Margin='5,0'"
+        L"       VerticalAlignment='Center'"
+        L"       InputScope='Number'"
+        L"       MaxLength='5' />"
+        L"     <TextBlock"
+        L"       Grid.Column='2'"
+        L"       Margin='2,2,2,2'"
+        L"       VerticalAlignment='Center'"
+        L"       Text='JS Engine (NYI):'/>"
+        L"     <ComboBox"
+        L"       x:Name='x_JsEngine'"
+        L"       Grid.Column='3'"
+        L"       Margin='2,2,2,2'"
+        L"       MinWidth='150'"
+        L"       HorizontalAlignment='Stretch'"
+        L"       VerticalAlignment='Center'>"
+        L"       <ComboBox.Items>"
+        L"         <ComboBoxItem Content='Chakra' />"
+        L"         <ComboBoxItem Content='Hermes' x:Name='x_engineHermes' />"
+        L"         <ComboBoxItem Content='V8' x:Name='x_engineV8' />"
+        L"       </ComboBox.Items>"
+        L"     </ComboBox>"
+        L"   </StackPanel>"
+        L""
+        L"   <Grid x:Name='RootElement'"
+        L"     Grid.Row='3'"
+        L"     Grid.ColumnSpan='2'"
+        L"     Background='{ThemeResource ApplicationPageBackgroundThemeBrush}'/>"
+        L" </Grid>";
+
+    Grid rootGrid = unbox_value<Grid>(Markup::XamlReader::Load(xamlString));
+    Window window = Window::Current();
+    window.Content(rootGrid);
+    window.Activate();
+
+    m_rootElement = rootGrid.FindName(L"RootElement").as<Grid>();
+
+    m_entryPointCombo = rootGrid.FindName(L"x_JavaScriptFilename").as<ComboBox>();
+    m_rootComponentNameCombo = rootGrid.FindName(L"x_ReactAppName").as<ComboBox>();
+    m_UseWebDebuggerCheckBox = rootGrid.FindName(L"x_UseWebDebuggerCheckBox").as<CheckBox>();
+    m_UseDirectDebuggerCheckBox = rootGrid.FindName(L"x_UseDirectDebuggerCheckBox").as<CheckBox>();
+    m_BreakOnFirstLineCheckBox = rootGrid.FindName(L"x_BreakOnFirstLineCheckBox").as<CheckBox>();
+    m_UseFastRefreshCheckBox = rootGrid.FindName(L"x_UseFastRefreshCheckBox").as<CheckBox>();
+    m_DebuggerPort = rootGrid.FindName(L"x_DebuggerPort").as<TextBox>();
+
+    m_DebuggerPort.Text(std::to_wstring(Host().InstanceSettings().DebuggerPort()));
+
+    // IsEditable is only supported on RS4 or higher
+    if (Windows::Foundation::Metadata::ApiInformation::IsApiContractPresent(
+            L"Windows.Foundation.UniversalApiContract", 6)) {
+      m_rootComponentNameCombo.IsEditable(true);
+      m_entryPointCombo.IsEditable(true);
+    }
+
+#if !defined(USE_HERMES)
```

Comment:
We're no longer passing this or the below as preprocessor definitions, so these will never be enabled.

Reply:
These definitions are pretty internal to RN. I don't think there's a programmatic way to check support currently. If playground is internal only, it might be fine to just always keep options enabled? IIRC we will fall back to Chakra if the engine isn't available (@tudorms can you confirm?). Depending on whether we assert, that could be a confusing dev experience.

Reply:
Actually Microsoft.ReactNative doesn't support plugging different JS engines at all.  So it might make sense just to remove this UI for now.

Reply:
That seems like we're losing the functionality to test different engines. Do we care about Hermes in UWP or should we drop that work?

---
In reply to: [406248271](https://github.com/microsoft/react-native-windows/pull/4534#discussion_r406248271) [](ancestors = 406248271)

Reply:
We definitely still care about that. If we wanted to reimplement it we would need to expose engine selection to Microsoft.ReactNative APIs and just forward to jsiEngineOverride through the ReactHost or whatever.